receivers/mqtt/config.go

```diff
@@ -16,14 +16,19 @@ const (
 )
 
 type Config struct {
-	BrokerURL          string `json:"brokerUrl,omitempty" yaml:"brokerUrl,omitempty"`
-	ClientID           string `json:"clientId,omitempty" yaml:"clientId,omitempty"`
-	Topic              string `json:"topic,omitempty" yaml:"topic,omitempty"`
-	Message            string `json:"message,omitempty" yaml:"message,omitempty"`
-	MessageFormat      string `json:"messageFormat,omitempty" yaml:"messageFormat,omitempty"`
-	Username           string `json:"username,omitempty" yaml:"username,omitempty"`
-	Password           string `json:"password,omitempty" yaml:"password,omitempty"`
-	InsecureSkipVerify bool   `json:"insecureSkipVerify,omitempty" yaml:"insecureSkipVerify,omitempty"`
+	BrokerURL            string                   `json:"brokerUrl,omitempty" yaml:"brokerUrl,omitempty"`
+	ClientID             string                   `json:"clientId,omitempty" yaml:"clientId,omitempty"`
+	Topic                string                   `json:"topic,omitempty" yaml:"topic,omitempty"`
+	Message              string                   `json:"message,omitempty" yaml:"message,omitempty"`
+	MessageFormat        string                   `json:"messageFormat,omitempty" yaml:"messageFormat,omitempty"`
+	Username             string                   `json:"username,omitempty" yaml:"username,omitempty"`
+	Password             string                   `json:"password,omitempty" yaml:"password,omitempty"`
+	QoS                  receivers.OptionalNumber `json:"qos,omitempty" yaml:"qos,omitempty"`
+	InsecureSkipVerify   bool                     `json:"insecureSkipVerify,omitempty" yaml:"insecureSkipVerify,omitempty"`
+	Retain               bool                     `json:"retain,omitempty" yaml:"retain,omitempty"`
+	TLSCACertificate     string                   `json:"tlsCACertificate,omitempty" yaml:"tlsCACertificate,omitempty"`
+	TLSClientCertificate string                   `json:"tlsClientCertificate,omitempty" yaml:"tlsClientCertificate,omitempty"`
+	TLSClientKey         string                   `json:"tlsClientKey,omitempty" yaml:"tlsClientKey,omitempty"`
```

Comment:
It would be great if we group these fields into a struct (similar to what upstream does). This schema will let us group fields in UI as well.
There are some problems with nested secrets right now but it should be fixed in https://github.com/grafana/grafana/pull/92035.

Reply:
Later we may reuse it in other integrations such as webhook